src/uu/sort/src/sort.rs

```diff
@@ -583,12 +783,22 @@ fn get_rand_string() -> String {
         .collect::<String>()
 }
 
-fn hash<T: Hash>(t: &T) -> u64 {
-    let mut s: XxHash64 = Default::default();
+fn get_hash<T: Hash>(t: &T) -> u64 {
+    let mut s: FnvHasher = Default::default();
     t.hash(&mut s);
     s.finish()
 }
 
+fn random_shuffle(a: &str, b: &str, x: String) -> Ordering {
```

Comment:
Would it be possible to have a test covering this line?

Reply:
Re: random shuffle and its output, I think the only predictable output we'd like to see is that duplicate lines are sorted together -- the concatenation of a random number and a line, which is then hashed. 

So, unfortunately, again here it may only be practical to do an internal unit test of the function itself and not its output.

Reply:
you could at least verify the size/length of the output, would be enough

Reply:
There could be a test to check that two consecutive runs produce different results and maybe that all lines from the input are present in the output

Reply:
All good ideas.  Thanks.  I've implemented them all.

Reply:
I want to make sure I got these right and clean them up a bit though.

Reply:
These seem to work.